src/features/Main/DeleteEnrollment/index.jsx

```diff
@@ -0,0 +1,162 @@
+import { useDispatch, useSelector } from 'react-redux';
+
+import {
+  updateStudentsTable,
+} from 'features/Students/data/slice';
+import { deleteEnrollment } from 'features/Main/data/api';
+
+const ERROR_MESSAGE = 'This student has reached a threshold of course activity that prevents unenrolled. If you wish to proceed, please contact support or your sales representative for assistance.';
+
+/**
+ * DeleteEnrollment allows an administrator to unenroll a student from a course.
+ * It handles confirmation, API interaction, error display, and feedback toast.
+ *
+ * Props:
+ * @param {string} studentEmail - The email of the student to unenroll.
+ * @param {string} courseId - The ID of the course from which to unenroll the student.
+ */
+const DeleteEnrollment = ({ studentEmail, courseId }) => {
+  const [message, setMessage] = useState('');
+  const [isSubmitting, setIsSubmitting] = useState(false);
+  const dispatch = useDispatch();
+  const students = useSelector((state) => state.students.table.data);
+
+  const [isError, openError, closeError] = useToggle(false);
+  const [isConfirmationOpen, openConfirmation, closeConfirmation] = useToggle(false);
+  const [showSuccessToast, openSuccessToast, closeSuccessToast] = useToggle(false);
+
+  const resetModals = () => {
+    setMessage('');
+    closeConfirmation();
+    closeSuccessToast();
+    closeError();
+    setIsSubmitting(false);
+  };
+
+  const handleDeleteEnrollment = async () => {
+    setIsSubmitting(true);
+    try {
+      const response = await deleteEnrollment(studentEmail, courseId);
+      const result = response?.data?.results?.[0];
+
+      if (result?.error) {
+        setMessage(ERROR_MESSAGE);
+        openError();
+      } else {
+        const updatedStudents = students.filter(
+          (student) => student.learnerEmail !== studentEmail,
+        );
+
+        dispatch(updateStudentsTable({
+          results: updatedStudents,
+          count: updatedStudents.length,
+        }));
+
+        setMessage('Enrollment deleted successfully.');
+        closeConfirmation();
+        openSuccessToast();
+      }
+    } catch {
+      setMessage('Unexpected error occurred. Please try again later.');
+      openError();
+    } finally {
+      setIsSubmitting(false);
+    }
+  };
+
+  return (
+    <>
+      <Dropdown.Item
+        className="text-truncate text-decoration-none custom-text-black"
+        onClick={openConfirmation}
+        data-testid="delete-enrollment"
+      >
+        Delete Enrollment
+      </Dropdown.Item>
+
+      <ModalDialog
+        title="Delete Enrollment"
+        isOpen={isConfirmationOpen}
+        onClose={closeConfirmation}
+        hasCloseButton
+        isFullscreenOnMobile
+      >
+        <ModalDialog.Header>
+          <ModalDialog.Title>
+            Delete Enrollment
+          </ModalDialog.Title>
+        </ModalDialog.Header>
+
+        <ModalDialog.Body>
+          You are attempting to unenroll a student.
```

Comment:
I think this text could be also defined in the library, so in the case we have to update the text we only have to updated there